66-irukandji/technical_specification.md

```diff
@@ -0,0 +1,66 @@
+# Schemapack (Irukandji)
+**Epic Type:** Implementation Epic
+
+Epic planning and implementation follow the
+[Epic Planning and Marathon SOP](https://docs.ghga-dev.de/main/sops/sop001_epic_planning.html).
+
+## Scope
+
+### Outline:
+This epic aims to make datapack navigation independent of its corresponding schemapack. Currently, since the relation class is not explicitly provided in the datapack, resolving target classes requires referring back to the schemapack. This epic introduces changes to eliminate this dependency, allowing the datapack to be self-contained for navigation.
+
+Additionally, this epic introduces configurable embedding depth in datapack denormalization. This ensures more control over how deeply related entities are embedded, enabling partial embedding instead of always including all related data at the highest level.
+
+### Included/Required:
+
+#### Example datapacks/schemapacks
+
+1. Update all the examples so that the datapack also has the rootClass, targetResources, rootResource and rootClass if it is specified in schemapack. 
+
+#### Datapack specification
+
+2. Change the datapack specification 
+   1. if there is  root resource, root resource has to exits
+   2. validating against datapack specification will rely on pydantic to ensure the root resource check. 
+3. The datapack specification will check if a datapack is valid or not. The validation plugins check if datapack is valid given a schemapack. 
```

Comment:
```suggestion
2. The datapack specification will check if a datapack is valid or not. The validation plugins check if datapack is valid given a schemapack. 
```

Not sure about the phrasing here, are those statements slightly redundant or does this mean that a datapack is checked for internal consistency first and then it is validated against a given schemapack?

Reply:
Yes. Rephrased